File: TASK_1-number_game.py
import random

def play_game():
    print("Welcome to the Number Game!")
    print("You need to guess the number I'm thinking of.")
    
    # Set the range for the random number
    lower_bound = 1
    upper_bound = 100
    max_attempts = 10  # Limit the number of attempts for the user
    total_score = 0
    
    play_again = "yes"
    
    while play_again.lower() in ["yes", "y"]:
        # Generate a random number
        random_number = random.randint(lower_bound, upper_bound)
        print(f"\nI'm thinking of a number between {lower_bound} and {upper_bound}. Can you guess it?")
        
        attempts = 0
        
        while attempts < max_attempts:
            try:
                # Prompt user for their guess
                guess = int(input(f"Attempt {attempts + 1}/{max_attempts}: Enter your guess: "))
            except ValueError:
                print("Invalid input! Please enter a number.")
                continue
            
            attempts += 1
            
            # Compare guess with the random number
            if guess == random_number:
                print(f"Congratulations! You guessed the correct number in {attempts} attempts.")
                total_score += max_attempts - attempts + 1  # Higher score for fewer attempts
                break
            elif guess < random_number:
                print("Too low! Try again.")
            else:
                print("Too high! Try again.")
        
        if attempts == max_attempts and guess != random_number:
            print(f"Sorry, you've run out of attempts! The correct number was {random_number}.")
        
        # Ask if the user wants to play again
        play_again = input("\nDo you want to play another round? (yes/no): ")
    
    # Display the final score
    print(f"\nGame over! Your total score is: {total_score}")

if __name__ == "__main__":
    play_game()

File: TASK-1 Numbergame.java
import java.util.Scanner;
import java.util.Random;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        boolean playAgain = true;
        int totalScore = 0;
        int round = 0;

        System.out.println("🎮 Welcome to the Number Guessing Game!");

        while (playAgain) {
            int numberToGuess = random.nextInt(100) + 1; // Generates number between 1 and 100
            int attempts = 0;
            int maxAttempts = 7;
            boolean guessedCorrectly = false;
            round++;

            System.out.println("\n🔁 Round " + round + ": I have picked a number between 1 and 100.");
            System.out.println("You have " + maxAttempts + " attempts. Good luck!");

            while (attempts < maxAttempts) {
                System.out.print("Enter your guess: ");
                int userGuess = scanner.nextInt();
                attempts++;

                if (userGuess == numberToGuess) {
                    int scoreThisRound = (maxAttempts - attempts + 1) * 10;
                    totalScore += scoreThisRound;
                    System.out.println("🎉 Correct! You guessed it in " + attempts + " attempts.");
                    System.out.println("🏆 You earned " + scoreThisRound + " points this round.");
                    guessedCorrectly = true;
                    break;
                } else if (userGuess < numberToGuess) {
                    System.out.println("📉 Too low! Try again.");
                } else {
                    System.out.println("📈 Too high! Try again.");
                }
            }

            if (!guessedCorrectly) {
                System.out.println("❌ Out of attempts! The number was: " + numberToGuess);
            }

            System.out.println("📊 Total Score: " + totalScore);
            System.out.print("Do you want to play another round? (yes/no): ");
            String response = scanner.next().toLowerCase();
            playAgain = response.equals("yes");
        }

        System.out.println("\n🎮 Game Over! Thanks for playing.");
        System.out.println("🎯 Final Score: " + totalScore);
        scanner.close();
    }
}
